### PR TITLE
eng-219

### PR DESCRIPTION
**Pull request by Connorbelez**
### TL;DR

Implemented effective-date ownership snapshot enforcement for dispersal entries to prevent post-calculation reroutes from affecting disbursement amounts.

### What changed?

Added audit metadata fields `ownershipSnapshotDate` and `reroutesAppliedCount` to dispersal entry calculation details. The `applyDealReroutes` function now returns the count of applied reroutes, and both fields are recorded during dispersal creation. Added defensive assertions in the disbursement bridge to validate calculation details exist and that entry amounts don't exceed distributable amounts.

### How to test?

Run the dispersal test suite to verify:
- Snapshot metadata is correctly recorded during dispersal creation
- Reroutes applied after dispersal calculation but before disbursement don't change the disbursement amount
- Bridge properly rejects entries with missing calculation details or amounts exceeding distributable limits

```bash
bun run test -- --run convex/dispersal/__tests__/
```

### Why make this change?

Ensures temporal consistency in the dispersal system by capturing ownership state at the time of calculation and preventing subsequent ownership changes from affecting already-calculated disbursement amounts. This maintains audit trails and prevents potential financial discrepancies.

## Summary by CodeRabbit

* **New Features**
  * Dispersal entries now record ownership snapshot dates and the count of reroutes applied.
  * Disbursement processing includes new validation gates to ensure calculation details are present and amounts do not exceed distributable values.

* **Tests**
  * Added tests validating snapshot metadata persistence and reroute scenarios.
  * Added tests covering disbursement validation failures for missing calculation details and excessive amounts.

* **Documentation**
  * New spec and task docs describing schema/backend changes and test plans.

### CODE REVIEW

File: specs/ENG-219/chunks/chunk-02-tests/tasks.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Chunk 02: Tests
+
+## Tasks
+- [ ] T-006: Edge-case test — reroute after dispersal, before disbursement
+- [ ] T-007: Bridge assertion tests — MISSING_CALCULATION_DETAILS + AMOUNT_EXCEEDS_DISTRIBUTABLE
+- [ ] T-008: Snapshot metadata tests — ownershipSnapshotDate + reroutesAppliedCount
+- [ ] T-009: Run codegen, typecheck, lint, and dispersal tests
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply specs/ENG-219/chunks/chunk-02-tests/tasks.md]: **Fix:** Updated T-006 through T-009 checkboxes to `[x]` to match the manifest's `complete` status.

File: specs/ENG-219/chunks/chunk-01-schema-backend/tasks.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Chunk 01: Schema & Backend Changes
+
+## Tasks
+- [ ] T-001: Add `ownershipSnapshotDate` and `reroutesAppliedCount` to `calculationDetailsValidator` in `convex/dispersal/validators.ts`
+- [ ] T-002: Add new fields to `CalculationDetails` interface in `convex/dispersal/types.ts`
+- [ ] T-003: Modify `applyDealReroutes()` return type + write new fields in `createDispersalEntries` handler
+- [ ] T-004: Add MISSING_CALCULATION_DETAILS assertion in `disbursementBridge.ts` processSingleDisbursement
+- [ ] T-005: Add AMOUNT_EXCEEDS_DISTRIBUTABLE assertion in `disbursementBridge.ts` processSingleDisbursement
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply specs/ENG-219/chunks/chunk-01-schema-backend/tasks.md]: **Fix:** Updated T-001 through T-005 checkboxes to `[x]` to match the manifest's `complete` status.

File: convex/dispersal/__tests__/createDispersalEntries.test.ts [hunk 3]
```diff
@@ -515,6 +515,70 @@ describe("createDispersalEntries", () => {
...
+
+			expect(result.created).toBe(true);
+			expect(result.entries).toHaveLength(2);
+
+			const persistedEntries = (
+				await t.run(async (ctx) =>
+					Promise.all(result.entries.map((entry) => ctx.db.get(entry.id)))
+				)
+			).filter((entry) => entry !== null);
+
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/dispersal/__tests__/createDispersalEntries.test.ts]: **Fix:** Replaced null-filtering with explicit non-null assertions. The array is now fetched without `.filter()`, followed by `expect(persistedEntries).toHaveLength(result.entries.length)` to catch any missing writes, and `expect(entry).not.toBeNull()` inside the loop to fail fast if any individual `ctx.db.get()` returns null. Optional chaining (`entry?.calculationDetails`) is used on the property accesses to satisfy the Biome linter (the preceding `not.toBeNull()` assertion already guards against null).
